network/requestTracker.go

```diff
@@ -379,6 +379,27 @@ func (rt *RequestTracker) Close() error {
 	return rt.listener.Close()
 }
 
+func (rt *RequestTracker) getWaitUntilEmptyChannel() <-chan struct{}{
+	done := make(chan struct{})
+
+	go func() {
+		for true {
+			select {
+			case <-time.After(100 * time.Millisecond):
+				rt.httpConnectionsMu.Lock()
+				if len(rt.httpConnections) == 0 {
+					done <- struct{}{}
+					rt.httpConnectionsMu.Unlock()
+					return
+				}
+				rt.httpConnectionsMu.Unlock()
+			}
+		}
+	}()
+
+	return done
```

Comment:
that looks like it would work, three nits:
1. make sure to run the test before invoking the go-routine. There is no reason to wait 100ms if it's already empty.
2. take advantage of close channel instead of writing to the channel.
3. pass in the sleep interval. I would choose a much smaller interval ( i.e. 5ms ).

Reply:
Thanks, updated. I'm also not sure why I went with select + time.After instead of a simple time.Sleep.